pallets/delegation/src/lib.rs

```diff
@@ -327,39 +306,33 @@ impl<T: Trait> Module<T> {
 		account: &T::AccountId,
 		delegation: &T::DelegationNodeId,
 		max_depth: u64,
-	) -> result::Result<bool, &'static str> {
-		if max_depth == 0 {
-			return Err("Reached max delegation search depth");
-		}
+	) -> Result<bool, DispatchError> {
+		// check for recursion anchor
+		ensure!(max_depth > 0, Error::<T>::MaxSearchDepthReached);
 
 		// check if delegation exists
-		let delegation_node = <error::Module<T>>::ok_or_deposit_err(
-			<Delegations<T>>::get(delegation),
-			Self::ERROR_DELEGATION_NOT_FOUND,
-		)?;
-		// check if the account is the owner of the delegation
+		let delegation_node =
+			<Delegations<T>>::get(delegation).ok_or(Error::<T>::DelegationNotFound)?;
+
+		// check if the given account is the owner of the delegation
 		if delegation_node.owner.eq(account) {
 			Ok(true)
 		} else if let Some(parent) = delegation_node.parent {
-			// recurse upwards in hierarchy
+			// recursively check upwards in hierarchy
 			Self::is_delegating(account, &parent, max_depth - 1)
 		} else {
-			// return whether the account is owner of the root
-			let root = <error::Module<T>>::ok_or_deposit_err(
-				<Root<T>>::get(delegation_node.root_id),
-				Self::ERROR_ROOT_NOT_FOUND,
-			)?;
+			// return whether the given account is the owner of the root
+			let root = <Root<T>>::get(delegation_node.root_id).ok_or(Error::<T>::RootNotFound)?;
 			Ok(root.owner.eq(account))
 		}
 	}
 
-	/// Revoke a delegation an all of its children
+	/// Revoke a delegation and all of its children
 	fn revoke(delegation: &T::DelegationNodeId, sender: &T::AccountId) -> DispatchResult {
```

Comment:
this is done recursively! 

[call to revoke in revoke_children](https://github.com/KILTprotocol/mashnet-node/pull/112/files#diff-c6838ae4ac2895550be01595658a64a2570e6a022ec8e62a23ad3ac62f8da94dR356)
[call to revoke_children in revoke](https://github.com/KILTprotocol/mashnet-node/pull/112/files#diff-c6838ae4ac2895550be01595658a64a2570e6a022ec8e62a23ad3ac62f8da94dR344)

tracked in KILTprotocol/ticket/issues/1068

Reply:
Good! I added a `FIXME` in 195109b.